dags/ingestion_neonat_cathydb.py

```diff
@@ -0,0 +1,109 @@
+POD_NAME = "raw-ingestion-neonat-cathydb"
+
+MAIN_CLASS = "bio.ferlab.ui.etl.red.raw.neonat.MainNeoCathydb"
+
+JAR = 's3a://spark-prd/jars/unic-etl-{{ params.branch }}.jar'
+
+default_args = {
+                'depends_on_past': False,
+                'start_date': datetime(2019, 5, 1),
+                'provide_context': True  # to use date of ingested data as input in main
+                }
+
+dag = DAG(
+    dag_id="ingestion_neonat_cathydb",
+    start_date=datetime(2019, 5, 1),
+    end_date=datetime(2023, 2, 14),
+    schedule_interval="@daily",
+    params={
+        "branch":  Param("UNIC-875", type="string"),
+        "version": Param("latest", type="string")
+    },
+    dagrun_timeout=timedelta(hours=2),
+    default_args=default_args,
+    is_paused_upon_creation=True,
+    catchup=True
+)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca_external_numeric", '{{ds}}'],  # {{ds}} input date
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag
+)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca_piicix_num", '{{ds}}'],
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag
+)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca_external_patient", '{{ds}}'],
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca_external_wave", '{{ds}}'],
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag
+)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca_piicix_sig", '{{ds}}'],
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag
+)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca_piicix_sig_calibre", '{{ds}}'],
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag
+)
+
+ingest_neonat_cathydb = SparkOperator(
+    task_id="ingest_neonat_cathydb",
+    name=POD_NAME,
+    arguments=["config/prod.conf", "default", "raw_icca.piicix_alertes", '{{ds}}'],
+    namespace=NAMESPACE,
+    spark_class=MAIN_CLASS,
+    spark_jar=JAR,
+    spark_config="medium-etl",
+    dag=dag
+)
```

Comment:
Il faut retourner les dépendances entre les tasks (https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#task-dependencies). On peut aussi rouler des tasks en parallèle si on les met entre [].

Reply:
les task 2, 5, 6 et 7 sont dependants du task 3 .... j'ai ajouté ça...
pour les tasks 1 et 4, ils n'ont pas de dépendances,

Reply:
On peut quand même rajouter 1, 3, 4 entre crochets pour indiquer qu'elles peuvent rouler en parallèle? Je crois que juste mettre `[ingest_neonat_cathydb1, ingest_neonat_cathydb3, ingest_neonat_cathydb4]` sur une autre ligne va fonctionner mais je ne suis pas certaine 🤔

Reply:
j'ai essayé ça mais j'ai un warning : statement seems to have no effect ...

Reply:
C'est juste un warning de l'IDE. Je crois que ça va quand même être pris en compte par Airflow car c'est fait comme ça dans les exemples dans leur doc... on déploiera et on verra 🙂